### PR TITLE
Update CI documentation to reflect the new CI

### PR DESCRIPTION
**Pull request by Mandur**
Co-authored-by: Nora Abi Akar <nora.abiakar@gmail.com>

# PR for issue #1900

## What is being addressed

Document the new CI model

### CODE REVIEW

File: docs/user-guide/ci-setup.md [hunk 1]
```diff
@@ -18,7 +18,11 @@ The [Build & Test CI pipeline][build-and-test-ci] runs the following tasks:
 ## LoRa E2E CI
 
 The [E2E CI pipeline][e2e-ci] is used to run end-to-end tests (implemented in
-the `LoRaWan.Tests.E2E` project). The pipeline runs the following tasks:
+the `LoRaWan.Tests.E2E` project). The pipeline runs tests using real LoRaWan hardware,
+it therefore require a VPN tunnel between the local hardware deployment and Azure. you
```
Comment (noraabiakar) [inline docs/user-guide/ci-setup.md:22]:
```suggestion
it therefore requires a VPN tunnel between the local hardware deployment and Azure. you
```

File: docs/user-guide/ci-setup.md [hunk 2]
```diff
@@ -33,6 +37,24 @@ the `LoRaWan.Tests.E2E` project). The pipeline runs the following tasks:
...
+### Pipeline authentication setup
+
+The pipeline uses the Github environment named *CI* and [OIDC tokens](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect) to authenticate to Azure. OIDC auth is enabled ONLY for this environement. All pipeline secrets are taken from a
+Keyvault. The only environment secrets needed at the time of writing are :
+
+- AZURE_CLIENT_ID : Required to set up the OIDC connection
+- AZURE_TENANT_ID : Required to set up the OIDC connection
+- AZURE_SUBSCRIPTION_ID : Required to set up the OIDC connection
+- AZURE_FUNCTIONAPP_PUBLISH_PROFILE : Required as the [Azure Function step doesn't support OIDC auth](https://github.com/Azure/functions-action/issues/153)
+- KEYVAULT_NAME : Required to indicate to which keyvault the pipeline should point
```
Comment (noraabiakar) [inline docs/user-guide/ci-setup.md:50]:
```suggestion
- KEYVAULT_NAME : Required to indicate which keyvault the pipeline should point to
```

File: docs/user-guide/ci-setup.md [hunk 3]
```diff
@@ -67,6 +89,52 @@ run previously failing tests, additional label must be added to the PR (e.g.
 label `1`). In this case the test labels already added to the PR will prevent
 those tests from being re-executed.
 
+### Connecting to the Environment
+
+CI operators can use an Azure VPN to [connect to the local CI using P2S](https://learn.microsoft.com/azure/vpn-gateway/vpn-gateway-howto-point-to-site-resource-manager-portal#connect), certificates and secrets are in the Keyvault
+
+### Manual steps to recreate the environment
+
+In addition of running the Bicep scripts, the following action needs to be manually executed:
```
Comment (noraabiakar) [inline docs/user-guide/ci-setup.md:98]:
```suggestion
In addition to running the Bicep scripts, the following action needs to be manually executed:
```

File: docs/user-guide/ci-setup.md [hunk 4]
```diff
@@ -67,6 +89,52 @@ run previously failing tests, additional label must be added to the PR (e.g.
...
+
+1. Create a service principal with permissions as described [above](#pipeline-authentication-setup)
+1. Create a new Github Environment (Or update the existing one) to reflect the client id of the new service principal
```
Comment (noraabiakar) [inline docs/user-guide/ci-setup.md:101]: Does that mean updating AZURE_CLIENT_ID, AZURE_TENANT_ID and AZURE_SUBSCRIPTION_ID?

**Review by noraabiakar — APPROVED** [review]
Looks great! I just a few typo corrections.

File: docs/user-guide/ci-setup.md (hunk 4 above)
Comment (Mandur) [reply docs/user-guide/ci-setup.md:101]: Zes, I will be more specific

Comment (Mandur) [reply docs/user-guide/ci-setup.md]: Let me know if this is better now

**Review by noraabiakar — APPROVED** [review]

File: docs/user-guide/ci-setup.md (hunk 4 above)
Comment (noraabiakar) [reply docs/user-guide/ci-setup.md]: Yes, perfect.

**Pull request merged**